### PR TITLE
Issue https://github.com/statOmics/msqrob2/issues/63

### PR DESCRIPTION
**Pull request by leopoldguyot**
Cf. issue:
https://github.com/statOmics/msqrob2/issues/63

### CODE REVIEW

File: R/msqrob.R [hunk 1]
```diff
@@ -108,16 +113,17 @@ msqrobLm <- function(y,
...
+                    rownames(vcovUnscaled) <- colnames(vcovUnscaled) <-  paramNames
                     vcovUnscaled[names(mod$coef), names(mod$coef)] <- msqrob2:::.vcovUnscaled(mod)
 
                     model <- list(
-                        coefficients = mod$coef,
-                        vcovUnscaled = .vcovUnscaled(mod),
+                        referencePresent = referencePresent,
+                        coefficients = coef,
+                        vcovUnscaled = vcovUnscaled,
                         sigma = sigma,
```
Comment (cvanderaa) [inline R/msqrob.R:127]: These changes need validation from @StijnVandenbulcke or @lievenclement

File: R/msqrob-utils.R [hunk 2]
```diff
@@ -182,9 +175,37 @@ checkFullRank <- function(modelMatrix) {
     return(qr(modelMatrix)$rank == ncol(modelMatrix))
 }
 
+
 ##### None exported functions from multcomp package is included here to
 ##### During R and Bioc checks
 
+propagateFalseStatus <- function(vectors, statuses) {
```
Comment (cvanderaa) [inline R/msqrob-utils.R:182]: Nice attempt to solve the interaction issue, I however can't fully wrap my head around what you are doing. A few comments on the function: 
- Could you document the internal function using standard comments so to briefly describe what the function does, and especially what are the expected argument inputs.  
- Be very careful with repeat to not allow for infinite loops, I personally prefer a `while()` loop although there is still is risk of infinite loops.
- I can't see unit tests, but maybe this is in progress

File: R/msqrob.R (hunk 1 above)
Comment (cvanderaa) [reply R/msqrob.R:129]: This has been validated during a live discussion.

**Comment by cvanderaa** [comment]
Let us know when your PR is done, I will then pull your branch and get my hands on your code to see if I can break it through some additional unit tests :smiling_imp:

File: R/msqrob-utils.R (hunk 2 above)
Comment (leopoldguyot) [reply R/msqrob-utils.R:179]: I added tests, commented the function and changed the code. https://github.com/statOmics/msqrob2/pull/64/commits/a97a4e5021bf49807b307cf9134156a32b261c0a

**Comment by cvanderaa** [comment]
This marks the end of Léopold's internship. The branch `issue63` is still work in progress, to be discussed in #63.

**Pull request merged**